app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
+    new_status = significant_status
+    if (previous_status != new_status) # we only care if something changed
+      if (!previous_status.success? && new_status.success?) || !new_status.success?
```

Comment:
We only want to fire the webhook when the `commit` transitions to `success?`. If the old status was `success` and a new status comes in that is also `success`, then we don't want to fire again. There's a test for that case.